File: Java/isPalindrome.java

/*
* @author Grace Hanson
* @brief Given an integer x, return true if x is a palindrome, and false otherwise.
* @date 3-27-2025
*/
class Solution {
    public boolean isPalindrome(int x) {
        String original = Integer.toString(x);
        String reversed = new StringBuilder(original).reverse().toString();
        return original.equals(reversed);

    }
}

File: Javascript/isPalindrome.js


/**
*  @author Grace Hanson
 * @param {number} x
 * @return {boolean}
 * @date 3-27-2025
 * @brief validate a palindrome via converting to string and splitting
 */
var isPalindrome = function (x) {
    //covnert x to a string
    const original = x.toString();
    const reversed = original.split('').reverse().join('');
    return original === reversed;
};